samza-core/src/main/java/org/apache/samza/container/grouper/stream/AllSspToSingleTaskGrouperFactory.java

```diff
@@ -73,62 +66,40 @@
       throw new SamzaException("Cannot process stream task with no input system stream partitions");
     }
 
-    if (isPassthroughJobCoordinator) {
-      final TaskName taskName = new TaskName(String.format("Task-%d", containerId));
+    processorNames.forEach(processorName -> {
+      // Create a task name for each processor and assign all partitions to each task name.
+      final TaskName taskName = new TaskName(String.format("Task-%s", processorName));
       groupedMap.put(taskName, ssps);
-    } else {
-      // Assign all partitions to each task
-      IntStream.range(0, taskCount).forEach(i -> {
-          final TaskName taskName = new TaskName(String.format("Task-%s", String.valueOf(i)));
-          groupedMap.put(taskName, ssps);
-        });
-    }
+    });
 
     return groupedMap;
   }
 }
 
 public class AllSspToSingleTaskGrouperFactory implements SystemStreamPartitionGrouperFactory {
+  private static final Logger LOG = LoggerFactory.getLogger(AllSspToSingleTaskGrouperFactory.class);
   private static final String YARN_CONTAINER_COUNT = "yarn.container.count";
-  private static final String PASSTHROUGH_JOB_COORDINATOR_FACTORY = PassthroughJobCoordinatorFactory.class.getName();
-  private static final String ZK_JOB_COORDINATOR_FACTORY = ZkJobCoordinatorFactory.class.getName();
 
   @Override
   public SystemStreamPartitionGrouper getSystemStreamPartitionGrouper(Config config) {
-    int taskCount;
-    int containerId = 0;
-    boolean isPassthroughJobCoordinator = false;
+    if (config.containsKey(TaskConfigJava.BROADCAST_INPUT_STREAMS)) {
+      throw new ConfigException("The job configured with AllSspToSingleTaskGrouper cannot have broadcast streams.");
+    }
+
     String jobCoordinatorFactoryClassName = config.get(JobCoordinatorConfig.JOB_COORDINATOR_FACTORY);
 
     if (jobCoordinatorFactoryClassName != null && !jobCoordinatorFactoryClassName.isEmpty()) {
-      if (jobCoordinatorFactoryClassName.equals(ZK_JOB_COORDINATOR_FACTORY)) {
-        taskCount = getNumProcessors(config);
-      } else if (jobCoordinatorFactoryClassName.equals(PASSTHROUGH_JOB_COORDINATOR_FACTORY)) {
-        if (config.get(JobConfig.PROCESSOR_ID()) == null) {
-          throw new ConfigException("Could not find " + JobConfig.PROCESSOR_ID() + " in Config!");
-        }
-        isPassthroughJobCoordinator = true;
-        containerId = config.getInt(JobConfig.PROCESSOR_ID());
-        taskCount = 1;
-      } else {
-        throw new ConfigException("AllSspToSingleTaskGrouperFactory is not yet supported with "
-            + jobCoordinatorFactoryClassName);
-      }
-    } else {
-      // must be cluster-based jobCoordinator (Yarn)
-      taskCount = config.getInt(YARN_CONTAINER_COUNT);
+      JobCoordinator jc = Util.
```

Comment:
There is still a case missing here: for cluster-based jobCoordinator, there is no configuration for the JC factory (yet-to-be-done refactoring work). Hence, a default method to create processorNames from the YARN_CONTAINER_COUNT is still needed now.

Reply:
Are you referring to the code that is already there on line 100-103 which gets the processorNames from YARN_CONTAINER_COUNT ?

Reply:
Ah, got it! I only checked the commit-after-last-review and somehow the diff was skipped. lgtm!